discord_quote/discord_quote/discord_quote.py

```diff
@@ -91,125 +92,295 @@ def me(ctx, *text : str):
-                output = {0}.replace(
-                    _quoter.group(0),
-                    "__via {0}__".format(ctx.message.author.name)
-                )
-            else:
-                # If the regex breaks, just forward the old message.
-                output = msg_.content
-        elif reply and quote:
-            log.info(log_msg(['formatting_quote', 'reply|quote']))
-            # Reply to a quotebot quote with a reply
-            output = '{0}\n**{1} [{2}] responded:** {3}'.format(
-                                clean_content,
-                                ctx.message.author.name,
-                                ctx.message.timestamp.strftime("%Y-%m-%d %H:%M:%S"),
-                                ' '.join(reply)
-                        )
-        else:
-            log.info(log_msg(['formatting_quote', 'reply|quote']))
-            output = (
-                '_heard_\n**{0} [{1}] said:** ```{2}```'
-                + '**{3} [{4}] responded:** {5}'
-            ).format(
-                     author,
-                     msg_.timestamp.strftime("%Y-%m-%d %H:%M:%S"),
-                     clean_content,
-                     ctx.message.author.name,
-                     ctx.message.timestamp.strftime("%Y-%m-%d %H:%M:%S"),
-                     ' '.join(reply)
+                      msg_id,
+                      ctx.message.channel.name,
+                      msg_.author.name,
+                      msg_.created_at.strftime("%Y-%m-%d %H:%M:%S"),
+                      ctx.message.author.name,
+                      msg_.clean_content]))
+
+        # Get, or create a webhook
+        hook = await _get_hook(ctx)
+
+        # Use WebHooks if possible
+        if hook:
+            payload = await webhook_quote(ctx, msg_, *reply)
+
+            # We need custom handling, so create a Webhook Adapter from our hook
+            await hook._adapter.execute_webhook(
+                payload={
+                    "content":payload,
+                    "username" : ctx.guild.me.name,
+                    "avatar_url": ctx.guild.me.avatar_url
+                }
             )
-        log.info(log_msg(['formatted_quote', ' '.join(reply)]))
-
-        yield from bot.say(output)
-
-        log.info(log_msg(['sent_message', 'quote', ctx.message.channel.name]))
 
-    except discord.errors.HTTPException:
-        log.warning(['msg_not_found', msg_id, ctx.message.author.mention])
+            log.info(log_msg(['sent_webhook_message',
+                              'quote',
+                              ctx.message.channel.name]))
+        else:
+            await bot_quote(ctx, msg_, *reply)
+    except discord.errors.HTTPException as e:
+        log.warning(['msg_not_found', msg_id, ctx.message.author.mention, e])
 
         # Return error if message not found.
-        yield from bot.say(("Quote not found in this channel ('{0}' "
+        await ctx.channel.send(("Quote not found in this channel ('{0}' "
```

Comment:
do we think its worth being consistent with fstring usage or is the nature of this case that a normal formatted string is more illustrative?